core/pkg/server/stream_init.go

```diff
@@ -147,6 +168,15 @@ func NewFileTransferManager(
 		fileTransferStats,
 	)
 
+	// Set proxy
+	fileTransferRetryClient.HTTPClient.Transport.(*http.Transport).Proxy = ProxyFn(settings.Proto.GetXProxies().GetValue())
+	// Set proxy authorization header, if present
+	if val, ok := settings.Proto.GetXExtraHttpHeaders().GetValue()["Proxy-Authorization"]; ok {
+		fileTransferRetryClient.HTTPClient.Transport.(*http.Transport).ProxyConnectHeader = http.Header{
+			"Proxy-Authorization": []string{val},
+		}
+	}
```

Comment:
after defining a `transport` variable initialized to a clean Transport:

```suggestion
	if val, ok := settings.Proto.GetXExtraHttpHeaders().GetValue()["Proxy-Authorization"]; ok {
		transport.ProxyConnectHeader.Add("Proxy-Authorization", val)
	}
```